src/Swift.Runtime/src/Swift/Runtime/InteropServices/SwiftMarshal.cs

```diff
@@ -15,22 +17,33 @@ public static class SwiftMarshal
     /// </summary>
     /// <typeparam name="T">The type of the value being marshaled</typeparam>
     /// <param name="value">The value to marshal</param>
-    /// <param name="swiftDest">the destination for marshaling</param>
+    /// <param name="swiftDestSpan">the destination for marshaling</param>
     /// <returns>A pointer to memory to pass in to Swift for marshaling. Note: this value may be different from the value passed in.</returns>
     /// <exception cref="NotSupportedException"></exception>
-    public static IntPtr MarshalToSwift<T>(T value, IntPtr swiftDest)
+    public static void MarshalToSwift<T>(T value, Span<byte> swiftDestSpan)
     {
         if (value is ISwiftObject swiftValue)
         {
-            return swiftValue.MarshalToSwift(new IntPtr(swiftDest));
+            swiftValue.MarshalToSwift(swiftDestSpan);
+            return;
         }
 
         var type = typeof(T);
         if (type.IsPrimitive || typeof(nint).IsAssignableFrom(type) || typeof(nuint).IsAssignableFrom(type))
         {
             unsafe
             {
-                return new IntPtr(MarshalPrimitiveToSwift(value, (void*)swiftDest));
+                var typeSize = Marshal.SizeOf<T>();
+                if (typeof(T) == typeof(bool))
```

Comment:
This check indicates some confusion. My guess is the `Marshal.SizeOf<bool>()` is returning 4 because that is the default marshalling behavior.

Reply:
When `bool` is marshalled, the memory is allocated as 
```csharp
byte* payload = stackalloc byte[(int)ElementSize];
Span<byte> span = new Span<byte>(payload, (int)ElementSize);
SwiftMarshal.MarshalToSwift(item, span);
```

where `ElementSize` is 1 on Swift side. It doesn't match marshalling size of `bool` on C# side.

Reply:
Right, this API shouldn't be used for this scenario.

Reply:
In general, you do not want to be using `Marshal.*` anywhere in the Swift support. `Marshal.*` are slow legacy APIs and they are unlikely to do what you want for Swift interop.